src/mrpro/data/traj_calculators/KTrajectoryCalculator.py

```diff
@@ -3,53 +3,93 @@
 from abc import ABC, abstractmethod
 
 import torch
+from einops import repeat
 
-from mrpro.data.enums import AcqFlags
-from mrpro.data.KHeader import KHeader
 from mrpro.data.KTrajectory import KTrajectory
 from mrpro.data.KTrajectoryRawShape import KTrajectoryRawShape
+from mrpro.data.SpatialDimension import SpatialDimension
 
 
 class KTrajectoryCalculator(ABC):
     """Base class for k-space trajectories."""
 
     @abstractmethod
-    def __call__(self, header: KHeader) -> KTrajectory | KTrajectoryRawShape:
+    def __call__(
+        self,
+        *,
+        n_k0: int,
+        k0_center: int | torch.Tensor,
+        k1_idx: torch.Tensor,
+        k1_center: int | torch.Tensor,
+        k2_idx: torch.Tensor,
+        k2_center: int | torch.Tensor,
+        encoding_matrix: SpatialDimension,
+        reversed_readout_mask: torch.Tensor | None = None,
+    ) -> KTrajectory | KTrajectoryRawShape:
         """Calculate the trajectory for given KHeader.
 
         The shapes of kz, ky and kx of the calculated trajectory must be
         broadcastable to (prod(all_other_dimensions), k2, k1, k0).
+
+        Not all of the parameters will be used by all implementations.
+
+        Parameters
+        ----------
+        n_k0
+            number of samples in k0
+        k1_idx
+            indices of k1
+        k2_idx
+            indices of k2
+        k0_center
+            position of k-space center in k0
+        k1_center
+            position of k-space center in k1
+        k2_center
+            position of k-space center in k2
+        reversed_readout_mask
+            boolean tensor indicating reversed redout
+        encoding_matrix
+            encoding matrix, describing the extend of the k-space coordinates
```

Comment:
```suggestion
            encoding matrix
```
The encoding matrix can be larger or smaller than the actual locations of the k-space sampling points